sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -407,53 +398,53 @@ object DateTimeUtils {
     }
   }
 
-  private def localTimestamp(microsec: SQLTimestamp, zoneId: ZoneId): LocalDateTime = {
+  private def localTimestamp(microsec: Long, zoneId: ZoneId): LocalDateTime = {
     microsToInstant(microsec).atZone(zoneId).toLocalDateTime
   }
 
   /**
    * Returns the hour value of a given timestamp value. The timestamp is expressed in microseconds.
    */
-  def getHours(microsec: SQLTimestamp, zoneId: ZoneId): Int = {
+  def getHours(microsec: Long, zoneId: ZoneId): Int = {
     localTimestamp(microsec, zoneId).getHour
   }
 
   /**
    * Returns the minute value of a given timestamp value. The timestamp is expressed in
    * microseconds.
    */
-  def getMinutes(microsec: SQLTimestamp, zoneId: ZoneId): Int = {
+  def getMinutes(microsec: Long, zoneId: ZoneId): Int = {
     localTimestamp(microsec, zoneId).getMinute
   }
 
   /**
    * Returns the second value of a given timestamp value. The timestamp is expressed in
    * microseconds.
    */
-  def getSeconds(microsec: SQLTimestamp, zoneId: ZoneId): Int = {
+  def getSeconds(microsec: Long, zoneId: ZoneId): Int = {
     localTimestamp(microsec, zoneId).getSecond
   }
 
   /**
    * Returns the seconds part and its fractional part with microseconds.
    */
-  def getSecondsWithFraction(microsec: SQLTimestamp, zoneId: ZoneId): Decimal = {
+  def getSecondsWithFraction(microsec: Long, zoneId: ZoneId): Decimal = {
     Decimal(getMicroseconds(microsec, zoneId), 8, 6)
   }
 
   /**
    * Returns seconds, including fractional parts, multiplied by 1000. The timestamp
    * is expressed in microseconds since the epoch.
    */
-  def getMilliseconds(timestamp: SQLTimestamp, zoneId: ZoneId): Decimal = {
+  def getMilliseconds(timestamp: Long, zoneId: ZoneId): Decimal = {
```

Comment:
shall we be consistent and use parameter name "microsec"?

Reply:
I would use `micros` to be consistent to functions names.

Reply:
And I am going to rename `date` to `days` when a parameter represents number of days since the epoch.